File: src/main/java/destiny/astrology/ZodiacDegreeDecoratorEnglish.java
/**
 * @author smallufo
 * Created on 2008/1/19 at 上午 5:59:26
 */
package destiny.astrology;

import org.jetbrains.annotations.NotNull;

import java.util.Locale;

public class ZodiacDegreeDecoratorEnglish implements ZodiacDegreeDecoratorIF {

  private double deg;

  @NotNull
  @Override
  public String getOutputString(double degree) {
    ZodiacSign sign = ZodiacSign.getZodiacSign(degree);
    this.deg = degree - sign.getDegree();

    return sign.toString(Locale.US) + getDeg() + "Deg " + getMin() + "Min " + getSec() + "Sec";
  }

  @NotNull
  @Override
  public String getSimpOutString(double degree) {
    ZodiacSign sign = ZodiacSign.getZodiacSign(degree);
    this.deg = degree - sign.getDegree();

    StringBuilder sb = new StringBuilder();
    if (getDeg() < 10)
      sb.append("0");
    sb.append(getDeg());
    sb.append(sign.getAbbreviation(Locale.US));
    if (getMin() < 10)
      sb.append("0");
    sb.append(getMin());
    return sb.toString();
  }

  private int getDeg() {
    return (int) deg;
  }

  private int getMin() {
    return (int) ((deg - getDeg()) * 60);
  }

  private double getSec() {
    return ((deg - getDeg()) * 60 - getMin()) * 60;
  }

}

File: src/main/java/destiny/astrology/ZodiacDegreeDecoratorEnglish.kt
/**
 * @author smallufo
 * Created on 2008/1/19 at 上午 5:59:26
 */
package destiny.astrology

import java.util.*

class ZodiacDegreeDecoratorEnglish : IZodiacDegreeDecorator {

  private var deg: Double = 0.toDouble()

  private val min: Int
    get() = ((deg - getDeg()) * 60).toInt()

  private val sec: Double
    get() = ((deg - getDeg()) * 60 - min) * 60

  override fun getOutputString(value: Double): String {
    val sign = ZodiacSign.getZodiacSign(value)
    this.deg = value - sign.degree

    return sign.toString(Locale.US) + " "  + getDeg() + "Deg " + min + "Min " + sec + "Sec"
  }

  override fun getSimpOutString(degree: Double): String {
    val sign = ZodiacSign.getZodiacSign(degree)
    this.deg = degree - sign.degree

    val sb = StringBuilder()
    if (getDeg() < 10)
      sb.append("0")
    sb.append(getDeg())
    sb.append(sign.getAbbreviation(Locale.US))
    if (min < 10)
      sb.append("0")
    sb.append(min)
    return sb.toString()
  }

  private fun getDeg(): Int {
    return deg.toInt()
  }

}
